python/beeai_framework/tools/python.py

```diff
@@ -0,0 +1,157 @@
+            input_files = (
+                list[python_files],
+                Field(
+                    description="""To access an existing file, you must specify it;
+                    otherwise, the file will not be accessible.
+                    IMPORTANT: If the file is not provided in the input, it will not be accessible."""
+                ),
+            )
+        self.input_schema = create_model(
+            "PythonToolInput",
+            language=(Language, Field(description="Use shell for ffmpeg, pandoc, yt-dlp")),
+            code=(str, Field(description="full source code file that will be executed")),
+            inputFiles=input_files,
+        )
+
+    def _create_emitter(self) -> Emitter:
+        return Emitter.root().child(
+            namespace=["tool", "python", "code_interpreter"],
+            creator=self,
+        )
+
+    async def _run(self, input: Any, options: ToolRunOptions | None, context: RunContext) -> PythonToolOutput:
+        async def get_source_code() -> Any:
+            if self.options and self.options.get("preprocess"):
+                response = await self.options["preprocess"]["llm"].create(
+                    {
+                        "messages": PromptTemplate(
+                            PromptTemplateInput(schema=UserPromptTemplateInput, template="{{input}}")
+                        ),
+                        "abortSignal": context.signal,
+                    }
+                )
+                return response.getTextContent().trim()
+            return input.code
+
+        async def call_code_interpreter(url: str, body: dict[str, Any], files: dict[str, io.BufferedReader]) -> Any:
+            headers = {"Accept": "application/json", "Content-Type": "application/json"}
+            async with httpx.AsyncClient() as client:
+                # response = await client.post(url, headers=headers, data=json.dumps(body), files=files)
+                response = await client.post(url, headers=headers, data=json.dumps(body), files=files)  # type:  ignore[arg-type]
+                response.raise_for_status()
+                return response.json()
+
+        if self.options and self.options.get("codeInterpreter") and self.options["codeInterpreter"].get("url"):
+            url = self.options["codeInterpreter"]["url"] + "/v1/execute"
+            prefix = "/workspace/"
+            unique_files = {}
+            if self.options and self.options.get("storage"):
+                for file in self.files:
+                    if file.filename in [python_file.value for python_file in input.inputFiles]:
+                        unique_files[file.filename] = file
+                self.options["storage"].upload(unique_files.values())
+
+            url = self.options["codeInterpreter"]["url"] + "/v1/execute"
+            files = {}
+            for file in unique_files.values():
+                files[prefix + file.filename] = open("localTmp/" + file.filename, "rb")  # noqa: ASYNC230,SIM115
```

Comment:
should we really be ignoring these lint issues?

Reply:
I couldn't find any other way to open multiple files and pass the file object to the API call.